### ISSUE TITLE
Switch component is not Accessible 

### ISSUE THREAD

**Issue by joe-s-avaya** [chakra-ui/chakra-ui#3480]
# 🐛 Bug report

> The Switch component is not properly accessible due to it breaking [Success Criterion 4.1.2](https://www.w3.org/TR/WCAG20-TECHS/F68); it is incapable of having a label

## 💥 Steps to reproduce

> This can be seen on the [docs page for Switch](https://chakra-ui.com/docs/form/switch) using the Siteimprove app, or just looking at the dev console

1. Go to [docs page for Switch](https://chakra-ui.com/docs/form/switch)
2. Open dev tools
3. View any of the Switch Elements
4. See that, because the Switch is wrapped in a Label, and does not accept an `aria-lable` attribute, it will always fail Success Criterion 4.1.2. You can use the [Siteimprove Chrome plugin](https://chrome.google.com/webstore/detail/siteimprove-accessibility/efcfolpjihicnikpmhnmphjhhpiclljc?hl=en-US) to see more details of this. Image attached for reference.

![switch-accessible-issue](https://user-images.githubusercontent.com/55896546/109441512-46ad4200-79ea-11eb-92b4-cec945441812.png)

## 💻 Link to reproduction

> Please create a minimal reproduction in CodeSandbox
>
>   [JavaScript Example](https://codesandbox.io/s/chakra-ui-switch-component-accessibility-issue-bpj9m?file=/src/App.js)

## 🧐 Expected behavior

> Like with the Checkbox, I expect one input to be associated with one label. In the case of the Switch, it's associated with _two_ labels when used as described by the current docs. 

## 🧭 Possible Solution

> Have the Switch containing element be a div instead of a label, and have the aria-label attribute (if passed) be applied to the input element. 

## 📝 Additional information

- aria-label example from codesandbox link
![codesandbox-chrome-console-example](https://user-images.githubusercontent.com/55896546/109442386-e66bcf80-79ec-11eb-971f-599813bb3238.png)

- I have not tested this with an actual screen reader, so this _may_ just be my plugin not understanding. But from what I understand, this **is** an accessibility issue.

**Comment by with-heart** [comment]
Ahh yeah, I see what you mean and have a hunch why it works this way. `Switch` is built with the same pattern as `Radio` and `Checkbox` where the outer wrapper of the component is a `label`. That reason that pattern makes sense for `Radio` and `Checkbox` because they expect and render `children`. `Switch` on the other hand doesn't, so it doesn't make sense to follow this pattern here.

I agree with your suggestion that we change the outer container of `Switch` to a `div`.

**Comment by dodas** [comment]
Why can I pass a label text as `children` to `<Checkbox>`, but not to `<Switch>`?
How is Switch different from Checkbox?

**Comment by with-heart** [comment]
@dodas There isn't much of a difference between them. It appears that `Switch` was originally a copy of `Checkbox` with modifications made to it, but that entirely doesn't make sense as `Switch` doesn't accept `children`. It could, and if it did then this would be fine, but as of right now, it doesn't. 

I think we have two options:

- Stick with the current API of not accepting `children` and use the solution suggested by @joe-s-avaya, where the container becomes a `div` and we make sure `aria-label` is passed to `input`
- Make `Switch` act like `Checkbox`, where it's expected that `children` is provided in order to label the element. This means the containing element remains a `label`, just like `Checkbox`.

I think the second method would have to be considered a breaking change, unless we were to basically support two different outputs depending on whether you pass `children` or not. The first output is the first solution where `Switch` is used like it is in the docs and the second output is the second solution.

Now that I say that, I'm actually interested in whether we could make that work. This could be a decent pattern for `Checkbox` and `Radio` as well, where they meet a11y requirements whether you pass `children` or not.

Thoughts?

**Comment by segunadebayo** [comment]
I think both options are viable. I most likely omitted the `children` part of the switch and I think we need to stick to a consistent API for input controls (Radio, Checkbox, Switch). 

Given that the children API didn't exist for the Switch, adding it would be a "minor", not a breaking change.

I'm working on a PR that resolves this accessibility issue.
